cirq-core/cirq/experiments/z_phase_calibration.py

```diff
@@ -0,0 +1,271 @@
+        n_circuits: The number of circuits to generate.
+        cycle_depths: The cycle depths to use.
+        random_state: The random state to use.
+        atol: Absolute tolerance to be used by the minimizer.
+        pool_or_num_workers: An optional multi-processing pool or number of workers.
+            A zero value means no multiprocessing.
+            A positivie integer value will create a pool with the given number of workers.
+            A None value will create pool with maximum number of workers.
+    Returns:
+        - An `XEBCharacterizationResult` object that contains the calibration result.
+        - A `pd.DataFrame` comparing the before and after fidelities.
+    """
+
+    pool: Optional['multiprocessing.pool.Pool'] = None
+    local_pool = False
+    if isinstance(pool_or_num_workers, multiprocessing.pool.Pool):
+        pool = pool_or_num_workers  # pragma: no cover
+    elif pool_or_num_workers != 0:
+        pool = multiprocessing.Pool(pool_or_num_workers)
+        local_pool = True
+
+    fids_df_0, circuits, sampled_df = parallel_xeb_workflow(
+        sampler=sampler,
+        qubits=qubits,
+        entangling_gate=two_qubit_gate,
+        n_repetitions=n_repetitions,
+        cycle_depths=cycle_depths,
+        n_circuits=n_circuits,
+        n_combinations=n_combinations,
+        random_state=random_state,
+        pool=pool,
+    )
+
+    if options is None:
+        options = xeb_fitting.XEBPhasedFSimCharacterizationOptions(
+            characterize_chi=True,
+            characterize_gamma=True,
+            characterize_zeta=True,
+            characterize_theta=False,
+            characterize_phi=False,
+        ).with_defaults_from_gate(two_qubit_gate)
+
+    p_circuits = [
+        xeb_fitting.parameterize_circuit(circuit, options, ops.GateFamily(two_qubit_gate))
+        for circuit in circuits
+    ]
+
+    result = xeb_fitting.characterize_phased_fsim_parameters_with_xeb_by_pair(
+        sampled_df=sampled_df,
+        parameterized_circuits=p_circuits,
+        cycle_depths=cycle_depths,
+        options=options,
+        fatol=atol,
+        xatol=atol,
+        pool=pool,
+    )
+
+    before_after = xeb_fitting.before_and_after_characterization(
+        fids_df_0, characterization_result=result
+    )
+
+    if local_pool:
+        assert isinstance(pool, multiprocessing.pool.Pool)
+        pool.close()
+    return result, before_after
+
+
+def calibrate_z_phases(
+    sampler: 'cirq.Sampler',
+    qubits: Optional[Sequence['cirq.GridQubit']] = None,
+    two_qubit_gate: 'cirq.Gate' = ops.CZ,
+    options: Optional[xeb_fitting.XEBPhasedFSimCharacterizationOptions] = None,
+    n_repetitions: int = 10**4,
+    n_combinations: int = 10,
+    n_circuits: int = 20,
+    cycle_depths: Sequence[int] = tuple(np.arange(3, 100, 20)),
+    random_state: 'cirq.RANDOM_STATE_OR_SEED_LIKE' = None,
+    atol: float = 1e-3,
+    pool_or_num_workers: Optional[Union[int, 'multiprocessing.pool.Pool']] = None,
```

Comment:
Please make single process the default and split to 2 arguments as suggested above for z_phase_calibration_workflow.

Reply:
@eliottrosenberg